### PR TITLE
Add 2.4.1 release notes

### PR DESCRIPTION
**Pull request by jovancacvetkovic**
Signed-off-by: Jovan Cvetkovic <jovanca.cvetkovic@gmail.com>

### Description
Add 2.4.1 release notes
 
### Issues Resolved
N/A
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-alerting-dashboards-plugin.release-notes-2.4.1.0.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+* Adjusted alerting plugin branch used by test workflows ([#390](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/390))
+
+### Documentation
+* Add 2.4.1 release notes ([#419](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/419))
+
+### Features
+* Align all flyout components to be overlay kind and add close button X in header ([#373](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/373))
+* Sort data by date for query visualization ([#385](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/385))
+* Avoid nesting search inside search when executing trigger condition for preview during create monitor ([#382](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/382))
+* Keep monitor table cell width flexible; Check for whether monitor is enabled when displaying related action ([#402](https://github.com/opensearch-project/alerting-dashboards-plugin/pull/402))
```
Comment (amsiglan) [inline release-notes/opensearch-alerting-dashboards-plugin.release-notes-2.4.1.0.md:23]: I think these three should go under bug fixes

**Review by amsiglan — APPROVED** [review]
Looks good except for one suggested change

File: release-notes/opensearch-alerting-dashboards-plugin.release-notes-2.4.1.0.md (hunk 1 above)
Comment (jovancacvetkovic) [reply release-notes/opensearch-alerting-dashboards-plugin.release-notes-2.4.1.0.md]: Moved to Bug Fixes.

**Comment by lezzago** [comment]
#420 merged

**Pull request closed without merging**